### PR TITLE
remove gometalinter

### PR DESCRIPTION
**Pull request by kyessenov**
gometalinter is deprecated and is about to be archived.
This PR removes all traces of its existence.

/assign @ozevren 

Signed-off-by: Kuat Yessenov <kuat@google.com>

### CODE REVIEW

File: mixer/tools/adapterlinter/main.go [hunk 1]
```diff
@@ -36,54 +38,79 @@ var invalidImportPaths = map[string]string{
...
+const (
+	dots = "/..."
+)
+
 func main() {
 	flag.Parse()
 	for _, r := range getReport(flag.Args()) {
-		reportErr(r)
+		// check if it skipped
+		if dat, err := ioutil.ReadFile(r.file); err == nil {
```
Comment (ozevren) [inline mixer/tools/adapterlinter/main.go]: Doesn't this increase the scope of nolint? Linter will start ignoring the whole file?

Comment (kyessenov) [reply mixer/tools/adapterlinter/main.go]: It looks for the line number from the report and only skips the particular line-bound report.

Comment (ozevren) [reply mixer/tools/adapterlinter/main.go]: Ah, the lines[] lookup below. It is not easy to see that. Can you add a comment to clarify?

Comment (kyessenov) [reply mixer/tools/adapterlinter/main.go]: Done

**Comment by kyessenov** [comment]
May get an approval @ozevren ?

File: tests/util/checker/checker.go [hunk 2]
```diff
@@ -53,6 +59,11 @@ func Check(paths []string, factory RulesFactory, whitelist *Whitelist, report *R
 
 // fileCheck checks a file using the given rules, and write to the given Report.
 func fileCheck(path string, rules []Rule, whitelist *Whitelist, report *Report) {
+	// skip over tests
+	if IgnoreTestLinterData && strings.Contains(path, "testlinter/testdata") {
```
Comment (ozevren) [inline tests/util/checker/checker.go]: This doesn't look like the right, final choice. A config plumbed in up top to control where to traverse is probably the better approach. Perhaps a bug and an associated TODO to handle this?

Comment (kyessenov) [reply tests/util/checker/checker.go]: Added a TODO. I agree that this needs further work, for a more consistent approach to skipping linter tests in a linter. It's probably the only useful capability of gometalinter that we relied upon, and now that it's gone, we need to find another way.

**Comment by kyessenov** [comment]
@geeknoid can you provide top-level approval?

**Pull request merged**